app/controllers/api/v1/transports_controller.rb

```diff
@@ -0,0 +1,77 @@
+module Api
+  module V1
+    class TransportsController < Api::V1::ApiController
+
+      skip_authorization_check only: :update_hook
+      load_and_authorize_resource :transport_order, parent: false, except: [:update_hook]
+
+      api :GET, '/v1/transports/providers', "List all GoodCity Tranports Options."
+      def providers
+        render json: TransportProvider.all.cached_json
+      end
+
+      api :POST, '/v1/transports/quote', "Get provider quote"
+      param :provider, String, desc: "Provider selected for transport"
+      param :vehicle_type, String, desc: "Transport vehicle-type"
+      param :offer_id, String, desc: "Id of the offer"
+      param :schedule_at, String, desc: "Scheduled time for delivery"
+      param :district_id, String, desc: "Id of the district"
+      def quote
+        order_price = TransportService.new(transport_params.to_h).quotation
+        render json: order_price
+      end
+
+      api :POST, '/v1/transports/book', "Book transport"
+      param :provider, String, desc: "Provider selected for transport"
+      param :vehicle_type, String, desc: "Transport vehicle-type"
+      param :offer_id, String, desc: "Id of the offer"
+      param :schedule_at, String, desc: "Scheduled time for delivery"
+      param :district_id, String, desc: "Id of the district"
+      param :pickup_contact_name, String, desc: "Contact Person Name"
+      param :pickup_contact_phone, String, desc: "Contact Person Mobile"
+      param :pickup_street_address, String, desc: "Pickup Address"
+      def book
+        order_info = TransportService.new(transport_params.to_h).book
+        render json: order_info
+      end
+
+      api :GET, '/v1/transports/:order_uuid', "Get GoodCity Tranport order details."
+      def show
+        order_info = TransportService.new({booking_id: params[:order_uuid], provider: transport_provider}).status
+        render json: order_info
+      end
+
+      api :POST, '/v1/transports/:order_uuid/cancel', "Cancel GoodCity Tranport order."
+      def cancel
+        order_info = TransportService.new({booking_id: params[:order_uuid], provider: transport_provider}).cancel
+        render json: order_info
+      end
+
+      api :POST, '/v1/transports/update_hook', "Webhook to update transport status"
+      def update_hook
```

Comment:
When implemented, we should use our own authorization check here. E.g. webhook can be `/api/v1/transports/update_hook?shared_key=ds23f934dfs&param1=df&param2=23`

Then check `shared_key` is as expected.